common/sphinx.c

```diff
@@ -669,3 +786,24 @@ u8 *unwrap_onionreply(const tal_t *ctx,
 		return tal_free(final);
 	return final;
 }
+
+u8 *sphinx_decompress(const tal_t *ctx, const u8 *compressed,
+		      struct secret *shared_secret)
+{
+	size_t compressedlen = tal_bytelen(compressed);
+	size_t prefill_size = TOTAL_PACKET_SIZE - compressedlen;
```

Comment:
Beware underflow here!  Must return NULL if compressedlen > TOTAL_PACKET_SIZE!

Reply:
Also, if compressedlen < HMAC_SIZE.

Reply:
Added asserts to check upper and lower size limits